infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -156,8 +158,11 @@ private synchronized void safeUpdateLayout(Layout layout) {
         // Cannot update with a null layout.
         if (layout == null) return;
 
-        // Update only if new layout has a higher epoch than the existing layout.
-        if (latestLayout == null || layout.getEpoch() > latestLayout.getEpoch()) {
+        // Update if new layout has a higher epoch than the existing layout.
+        // Or if the cluster ID is different.
+        if (latestLayout == null ||
+                layout.getEpoch() > latestLayout.getEpoch() ||
+                !layout.getClusterId().equals(latestLayout.getClusterId())) {
```

Comment:
Change the layout any time cluster ID doesn't match?...That looks very permissive.